src/workloads/query/GroupStagesOnComputedFields.yml

```diff
@@ -0,0 +1,176 @@
+      active: {^Parameter: {Name: "active", Default: [3]}}
+      nopInPhasesUpTo: &NopInPhasesUpTo {^Parameter: {Name: "nopInPhasesUpTo", Default: *MaxPhases}}
+      pipeline: [{$addFields:
+                          {"hourDiff": {$dateDiff: {"startDate": "$time", "endDate": {^Date: "2022-01-04"}, "unit": "hour"}}}},
+                          {$match: {"hourDiff": {$lte: 12}}},
+                          {$group:
+                            {"_id" :
+                              { "symbol" : "$symbol" ,
+                                "time": { "$dateTrunc" : {"date": "$time", "unit": "minute", "binSize" : 60}}
+                              },
+                              "open" : {"$first" : "$price"},
+                              "close" : {"$last" : "$price"}}
+                          },
+                          {
+                              $addFields : {
+                                  "diffPercentage" : {$round : [{$multiply : [100, { $divide : [{$subtract : ["$close", "$open"]}, "$open"] }]}, 2]}
+                              }
+                          },
+                          {
+                            $group: {
+                                  "_id" : "$_id.time",
+                                  "topN": {
+                                      $topN: {
+                                          "output": ["$_id.symbol", "$diffPercentage", "$close"],
+                                          "sortBy" : { "diffPercentage" : -1 },
+                                          "n": 3
+                                      }
+                                  },
+                                  "bottomN": {
+                                      $bottomN: {
+                                          "output": ["$_id.symbol", "$diffPercentage", "$close"],
+                                          "sortBy" : { "diffPercentage" : 1 },
+                                          "n": 3
+                                      }
+                                  },
+                              }
+                          },
+                          {
+                              $sort : {
+                                  "_id" : -1
+                              }
+                          }]
+
+GroupStageAfterMatchOnDateComputation: &GroupStageAfterMatchOnDateComputation
+  LoadConfig:
+    <<: *loadConfig
+    Parameters:
+      name: GroupStageAfterMatchOnDateComputation
+      active: {^Parameter: {Name: "active", Default: [4]}}
+      nopInPhasesUpTo: *NopInPhasesUpTo
+      pipeline: [{$match: {
+                          "symbol": "string050",
+                          "time": {
+                            $gte: "2022-01-03"
+                          }
+                          }
+                  },
+                  {$group:
+                    {"_id" : {
+                        $dateTrunc: {
+                          "date": "$time",
+                          "unit": "day"
```

Comment:
If you have time, I'd be curious to see how quantizing by minute would change our results. Having more groups might change which bottlenecks we see in each of the implementations. The characteristics might be different enough to justify two benchmarks.

Reply:
I tried this. For the docs returned, quantizing by day returns 3 documents, and quantizing by minute returns 2937 documents. Perf results for this experiment are [here](https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?comparison_id=5392c9c9-1c51-4bd2-ad09-9d6917120a0f&task_filter=group_stages_on_computed_fields&test_filter=GroupStageOnDateComputationAfterMatch.GroupStageOnDateComputationAfterMatch&test_filter=GroupStageAfterSortAndAddFields.GroupStageAfterSortAndAddFields&test_filter=MultipleGroupStagesOnComputedFields.MultipleGroupStagesOnComputedFields&measurement_filter=OperationThroughput&percent_filter=0%7C%7C100&z_filter=0%7C%7C10). Should I add this as a separate query or try seconds as well?